### PR TITLE
initialized backend

### PR DESCRIPTION
**Pull request by Pradeep-Kumar-Rebbavarapu**
* Fix Issue #2

### CODE REVIEW

**Comment by gjain-7** [comment]
1. Reference the issue in PR message. Follow this https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests.
2. Put the tasks (in the issue)  with **marked checkboxes**.

**Review by gjain-7 — CHANGES_REQUESTED** [review]
1. Reference the issue in PR message. Follow https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests.
2. Use a small yet meaningful name for PR (like `initializing-backend` or something).
3. Put the tasks (in the issue)  with **marked checkboxes**.
4. If possible, squash and rebase your commits, to include meaningful names and messages or just create a new pr. Follow [wiki](https://github.com/Web-Team-IITI-Gymkhana/tpc-backend/wiki/Contributing#writing-commit-messages).
5. Don't use npm, use yarn.
6. Some config files seem to be deleted while others are accidentally added. Please fix it.
7. Update versions in `test.yml` (node version, git-checkout, setup-node)
8. Use clean slate install https://stackoverflow.com/a/69944063/19948093 instead of plain `yarn install` in the pipeline.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
```
Comment (gjain-7) [inline .github/workflows/test.yml:14]: update to `actions/checkout@v2`

Comment (Pradeep-Kumar-Rebbavarapu) [reply .github/workflows/test.yml:14]: made the required changes

**Comment by Pradeep-Kumar-Rebbavarapu** [comment]
* Also done with the issue #5

**Review by gjain-7 — CHANGES_REQUESTED** [review]
1. Remove the `dist` from the commit
3. Use a small yet meaningful name for PR (like `initializing-backend`).
4. Add the task list (from the issue) in the PR comment with the `checkboxes marked`.
In your case it would be 
```md
- [x] Create a sample route that returns the text `'hello world'` and its corresponding unit test to confirm its functionality.
- [x] set up a gitHub action to run automated test on each pr.
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
```
[the remark this answers is not in the document]
Comment (Pradeep-Kumar-Rebbavarapu) [reply .github/workflows/test.yml]: done

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
```
Comment (gjain-7) [inline .github/workflows/test.yml:17]: I am sorry I mentioned 
> update to actions/setup-node@v4

but the latest version is [3](https://github.com/actions/setup-node/tree/main). So please change it to 3

**Review by gjain-7 — CHANGES_REQUESTED** [review]

**Comment by Pradeep-Kumar-Rebbavarapu** [comment]
**Checklist:**
- [X] Create a sample route that returns the text 'hello world' and its corresponding unit test to confirm its functionality.
- [X] set up a gitHub action to run automated test on each pr.

**Comment by akashg3627** [comment]
Please use `yarn` to manage packages, so delete the`package-lock.json` and then run `yarn`.

**Comment by akashg3627** [comment]
I am concerned about the folder structure a bit.
We should not divide things into admin, recruiter, student in backend -> there should be folder to manage each entity -> and keep the `tests` as separate folder outside the `src` folder.

File: .github/workflows/test.yml (hunk 3 above)
Comment (Pradeep-Kumar-Rebbavarapu) [reply .github/workflows/test.yml]: done changed it to 3

**Comment by Pradeep-Kumar-Rebbavarapu** [comment]
removed package-lock.json on the latest commit

**Comment by akashg3627** [comment]
It would be really good, if we can find some opensource project or doc on folder structure and set it as the rule. Write a `wiki` on same, so that its easy to grab for any new contributor.
for eg. [typescript-starter](https://github.com/nestjs/typescript-starter/tree/master)

**Comment by akashg3627** [comment]
Some suggested projects:
- https://github.com/CatsMiaow/nestjs-project-structure/tree/master

**Review by akashg3627 — DISMISSED** [review]
Please merge this proceed to adding the db and all. Use sequelize as orm.

**Comment by akashg3627** [comment]
Before merging commit the `yarn.lock` file

**Comment by Pradeep-Kumar-Rebbavarapu** [comment]
added yarn.lock file

**Review by gjain-7 — APPROVED** [review]

**Pull request merged**